### ISSUE TITLE
Re-consideration `poke` for Deposits 

### ISSUE THREAD

**Issue by kianenigma** [paritytech/polkadot-sdk#5591]
Many pallets are collecting deposits to prevent spam. Some hard-code values that indicate what that deposit is, some use the newer `Consideration` API. Some efforts exist to migrate more pallets to the new API: https://github.com/paritytech/polkadot-sdk/issues/226

In both cases, the pallets should be capable of adjusting the deposit in a lazy manner. As in, if the "_deposit formula_" (regardless of how it is expressed) changes, we can re-adjust it. 

This should be done lazily, via a permissionless `fn poke()` transaction that is free in the case of a successful adjustment, and a paid noop otherwise. 

Pallets in scope of migration:
- [x] pallet-indices #7587 
- [x] pallet-multisig: #7700 

- [x] pallet-proxy: #7801 
- [x] pallet-society: #7833
Deposit reserved here: [1](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/society/src/lib.rs#L851)
- [x] pallet-bounties: #8382 
Deposit reserved here: [1](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/bounties/src/lib.rs#L578), [2](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/bounties/src/lib.rs#L985)
- [x] pallet-recovery: #7882 
Deposit reserved here: [1](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/recovery/src/lib.rs#L496), [2](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/recovery/src/lib.rs#L539)

**Comment by xlc** [comment]
Yeah we should have lot more utility pallets to reduce number of non-business logic code in each pallet.
pallet-parameters is one improve that eliminates the needs to implement parameter handling logics. We could have a similar pallet handles all the deposits and poke etc.

**Comment by kianenigma** [comment]
good name: `fn reconisder(origin)`.

**Comment by ggwpez** [comment]
Leaning demo can be pallet indices: https://github.com/polkadot-fellows/runtimes/pull/577#discussion_r1941920392

**Comment by muharem** [comment]
for bounties pallets you probably need the poke call only for this https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/bounties/src/lib.rs#L985
the rest is reserve based on the curator fee, not the storage usage fee

**Comment by UtkarshBhardwaj007** [comment]
> for bounties pallets you probably need the poke call only for this https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/bounties/src/lib.rs#L985 the rest is reserve based on the curator fee, not the storage usage fee

Similarly for child bounties, the only deposit reserved is for curators which is calculated based on a multiplier and not storage. So this won't need `poke_deposit` as well. I have the last PR out for this issue for `pallet-bounties`.
